clusterloader/framework/util.go

```diff
@@ -79,3 +106,248 @@ func CreatePods(f *framework.Framework, appName string, ns string, labels map[st
+	if podTuning != "" {
+		// Iterate through defined tuningSets
+		for _, ts := range tuningSets {
+			// If we have a matching tuningSet keep it
+			if ts.Name == podTuning {
+				tuning = &ts
+				return
+			}
+		}
+		framework.Failf("No pod tuning found for: %s", podTuning)
+	}
+	return nil
+}
+
+// Server is the webservice that will syncronize the start and stop of Pods
+func Server(c *PodCount) error {
+	http.HandleFunc("/start", handleStart(startHandler, c))
+	http.HandleFunc("/stop", handleStop(stopHandler, c))
+
+	server := &http.Server{Addr: ":8081", Handler: nil}
+
+	ln, err := net.Listen("tcp", server.Addr)
+	if err != nil {
+		return err
+	}
+
+	go server.Serve(ln)
+	fmt.Println("Listening on port", server.Addr)
+	select {
+	case <-c.Shutdown:
+		fmt.Println("Shutdown server")
+		ln.Close()
+		return err
+	}
+}
+
+func handleStart(fn http.HandlerFunc, c *PodCount) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		c.Started++
+		fn(w, r)
+		fmt.Printf("Started pods: %d, Stopped pods: %d\n", c.Started, c.Stopped)
+	}
+}
+
+func startHandler(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintln(w, "Hello")
+}
+
+func handleStop(fn http.HandlerFunc, c *PodCount) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		c.Stopped++
+		fn(w, r)
+		fmt.Printf("Started pods: %d, Stopped pods: %d\n", c.Started, c.Stopped)
+		if c.Stopped == c.Started && c.Stopped > 0 {
+			c.Shutdown <- true
+		}
+	}
+}
+
+func stopHandler(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintln(w, "Goodbye")
+}
+
+// firstLowercase conversts the first letter of a string to lowercase.
+func firstLowercase(s string) string {
+	a := []rune(s)
+	a[0] = unicode.ToLower(a[0])
+	return string(a)
+}
+
+// convertVariables takes our loaded struct and converts it into a map[string]string.
+func convertVariables(params ParameterConfigType) map[string]string {
+	m := structs.Map(params)
+	values := make(map[string]string)
+	for k, v := range m {
+		k = firstLowercase(k)
+		if v != 0 && v != "" {
+			if _, ok := v.(int); ok {
+				values[k] = strconv.Itoa(v.(int))
+			} else {
+				values[k] = v.(string)
+			}
+		}
+	}
+	return values
+}
+
+// InjectConfigMap modifies the pod struct and replaces the environment variables.
+func InjectConfigMap(f *framework.Framework, ns string, vars ParameterConfigType, config v1.Pod) string {
+	configMapName := ns + "-configmap"
+	freshConfigVars := convertVariables(vars)
+	dirtyConfigVars := getClusterData(f, freshConfigVars)
+	configMap := newConfigMap(ns, configMapName, dirtyConfigVars)
+	framework.Logf("Creating configMap %v in namespace %v", configMap.Name, ns)
+	var err error
+	if configMap, err = f.ClientSet.Core().ConfigMaps(ns).Create(configMap); err != nil {
+		framework.Failf("Unable to create test configMap %s: %v", configMap.Name, err)
+	}
+
+	for i, envVar := range config.Spec.Containers[0].Env {
```

Comment:
I don't know how this will work. My gut feeling is that kubelet should handle this. @yujuhong